sway-core/src/engine_threading.rs

```diff
@@ -52,11 +52,26 @@ impl Engines {
 
     /// Removes all data associated with `source_id` from the engines.
     /// It is intended to be used during garbage collection to remove any data that is no longer needed.
+    ///
+    /// It will also clear the associated autogenerated file for the `source_id` parameter.
     pub fn clear_module(&mut self, source_id: &sway_types::SourceId) {
         self.type_engine.clear_module(source_id);
         self.decl_engine.clear_module(source_id);
         self.parsed_decl_engine.clear_module(source_id);
         self.query_engine.clear_module(source_id);
+
+        // Check if `source_id` has an associated autogenerated file
+        // and clear it
```

Comment:
should we add this logic to the `clear_program` function above. I don't think it's getting used anywhere from memory at the moment but if we are keeping it around it probably makes sense to add this logic there as well.

Reply:
I don´t think we need it because the auto-generated pseudo-file is associated with the same "program_id" as the original "source_id". Thus, clearing the program will clear them both.

Reply:
ok cool thanks for clarifying.